pkg/vault/contracts/VaultCommon.sol

```diff
@@ -332,10 +332,7 @@ abstract contract VaultCommon is IVaultEvents, IVaultErrors, VaultStorage, Reent
         }
 
         // Still has to be a valid percentage, regardless of what the pool defines.
-        if (
-            swapFeePercentage > ISwapFeePercentageBounds(pool).getMaximumSwapFeePercentage() ||
-            swapFeePercentage > FixedPoint.ONE
```

Comment:
This is in case the pool returns > 1. Even if not malicious, it could have some kind of calculation (maybe with an external call somewhere) with a bug in it that causes it to return invalid values if we just blindly take whatever the pool returns.

Ahhh - but this is checked later in `PoolConfigLib.setStaticSwapFeePercentage`, so the check here is now redundant. (I don't think it was when this was written.) `InvalidPercentage` is also a more specific error message (though the "too high" one isn't wrong).

I suggest adding a comment on that library call saying it validates the percentage.